xds/internal/client/client_watchers_rds.go

```diff
@@ -40,8 +40,20 @@ type HeaderMatcher struct {
 	PresentMatch *bool       `json:"presentMatch,omitempty"`
 }
 
+// Route represents route with matchers and action.
+type Route struct {
+	Path, Prefix, Regex *string
+	Headers             []*HeaderMatcher
+	Fraction            *uint32
+	Action              map[string]uint32 // action is weighted clusters.
+}
+
 type rdsUpdate struct {
+	// weightedCluster is only set when routing is disabled (not enabled) by env
+	// variable.
 	weightedCluster map[string]uint32
+
+	routes []*Route
```

Comment:
Would it make sense to fold the existing `weightedCluster` field into a `Route` as well? So, in that case the xdsResolver would always get an `rdsUpdate` with a slice of `Route`s. And in the case where xdsRouting balancer is not to be used, it will expect a single entry with the slice with no matchers, and only an action containing the weightedClusters.